---
layout: page
---

# Introduction to STOM HTS Workshop 2018

<hr>
<br>

**Overview:** This hands-on sessions introduce students of STOM HTS
workshop to computational tools, techniques, and best practices for
analyzing high-throughput sequence data in population scale.

**Description:** 
The dramatic advance of high-throughput sequencing technologies in the last decade has produced
tremendous amount genomic, transcriptomic, and epigenomic sequence data at an unprecedented
scale. To investigate the impact of naturally occurring genetic variants on disease traits and, it is
important to understand how to properly analyze many sequenced genomes together in a harmonized
manner so that powerful statistical methods can be applied. In this short course, we aim to learn the
key concepts and knowledge to design whole-genome or whole-exome sequencing studies for
genetic mapping in population-based studies. We will focus on practical knowledge and hands on
experiences in analyzing sequence data, including alignment, quality control, calling genetic variants
such as SNPs, Indels, and structural variants. We will use publicly available RNA-seq datasets as
examples of intermediate phenotypes and learn how to perform eQTL analysis between sequence
genomes and transcriptomes as examples of multi-omics sequence analysis at population-scale.

**Audience:** This course is intended for researchers who are actively engaged in
genomics research and interested beginners, including laboratory scientists and clinicians with a
basic quantitative background. Ideally, participants are expected to have some basic knowledge of
human genetics (such as Mendelian inheritance), core statistical principles (such as p-values), and
basic UNIX skills (such as free contents material at
[https://www.codecademy.com/learn/learn-the-command-line](https://www.codecademy.com/learn/learn-the-command-line))

**Requirements:** Participants must bring a laptop with specific [software installed]({{ site.baseurl }}/setup/).

**When:** July 25-26 (2018), 9:20 AM - 5:10 PM 

**Where:** 25-411, Seoul National University, Kwanak-gu, Seoul, Korea.

<br>
<hr>
<br>

#### [Day 1. Analysis of High-throughput Sequence Reads]({{ site.baseurl }}/day1/) 

| Session | Time           | Topics                   | 
| :-----: |:--------------:| :----------------------- | 
| I       | 9:20-10:50 AM  | **Design and analysis of sequencing studies in population scale** | 
|         | 10:50-11:00 AM | Coffee Break             | 
| II      | 11:00-12:30 PM | **Alignment and quality control of sequenced genomes**       | 
|         | 12:30-2:00 PM  | Lunch                    | 
| III     | 2:00-3:30 PM   | **Calling short variants (SNPs and Indels) from sequence reads**    | 
|         | 3:30-3:40 PM   | Coffee Break             | 
| IV      | 3:40-5:10 PM   | **Calling structural variants (large deletions and CNVs) from sequence reads**   | 

<br>
<hr>
<br>

#### [Day 2. Population-scale analysis of DNA and RNA sequence data]({{site.baseurl}}/day2/)

| Session | Time           | Topics                   | 
| :-----: |:--------------:| :----------------------- | 
| I       | 9:20-10:50 AM  | **Variant annotation, filtering, and data harmonization** | 
|         | 10:50-11:00 AM | Coffee Break             | 
| II      | 11:00-12:30 PM | **Haplotype-aware analysis of genomic sequence reads**       | 
|         | 12:30-2:00 PM  | Lunch                    | 
| III     | 2:00-3:30 PM   | **Introduction to RNA-seq experiments and eQTL analysis**    | 
|         | 3:30-3:40 PM   | Coffee Break             | 
| IV      | 3:40-5:10 PM   | **Population-scale analysis of single-cell RNA-seq data**   | 

<br>
<hr>
<br>

<!--## Relevant Resources 
**BIOINF-575**: Programing Lab in Bioinformatics  
	
**Software Carpentry**: Occasional Workshops  
(Non planned for this year at UM unfortunately) 
<http://software-carpentry.org> -->

<!--- Uncomment at end of course...
Add more courses when we find them.
-->


